Celeste.Mod.mm/Patches/LevelData.cs

```diff
@@ -108,5 +125,25 @@ we found
                 throw new Exception($"Too few matches for HasAttr(\"tag\"): {matches}");
             }
         }
+
+        public static void PatchLevelDataSpawnpointLoader(ILContext context, CustomAttribute attrib) {
+            MethodDefinition m_LevelDataCheckForDefaultSpawn = context.Method.DeclaringType.FindMethod("CheckForDefaultSpawn");
+            TypeReference t_Vector2 = MonoModRule.Modder.FindType("Microsoft.Xna.Framework.Vector2").Resolve();
+            t_Vector2 = MonoModRule.Modder.Module.ImportReference(t_Vector2);
+            VariableDefinition v_spawnCoords = new VariableDefinition(t_Vector2);
+            context.Body.Variables.Add(v_spawnCoords);
+
+            ILCursor cursor = new ILCursor(context);
+
+            // call CheckForDefaultSpawn with checkpoint element and coordinates
+            cursor.GotoNext(instr => instr.MatchLdstr("player"));
+            cursor.GotoNext(MoveType.After, instr => instr.MatchNewobj("Microsoft.Xna.Framework.Vector2"));
+            cursor.Emit(OpCodes.Stloc, v_spawnCoords);
+            cursor.Emit(OpCodes.Ldarg_0);
+            cursor.Emit(OpCodes.Ldloc, 9);
```

Comment:
I think that the local variable number can change if the method gets more variables added to it? Would probably be more stable if you grab the one you need earlier using e.g. `instr.MatchLdloc(out int num)`

Reply:
I mean you're adding a new variable into the middle lol. I guess I'd have to check the IL to see how it changes